File: workUndone/Suite15/OpenFlight_API/samples/scripts/egeyepoint1.py
##
##
##  Sample file: egeyepoint1.py
##
##  Program functions:	Opens a database from the command line.
##     Reads in eyepoint records from the database
##	  Prints out all eyepoints, or only active eyepoints in the database
##	  Closes the database.
##
##  API functions used:
##     mgInit(), mgExit(), mgGetLastError(),
##	  mgGetAttList(), mgGetEyePoint(), mgSetEyePoint(),
##	  mgOpenDb(), mgCloseDb(), 
##
##

import sys

# import OpenFlight API module
from mgapilib import *
from printeye import PrintEyePoint

def main ():
   # whether to write out eyepoints that haven't been saved 
	writeNonActives = MG_FALSE

   # check for proper arguments 
	if len(sys.argv) == 3 and sys.argv[2] == "-a":
		writeNonActives = MG_TRUE
	elif len(sys.argv) < 2:
		print "\nUsage: %s <input_db_filename> [-a]\n" % (sys.argv[0])
		print "   Reads database: <input_db_filename>\n"
		print "   Prints eyepoints from database\n"
		print "   -a : Print all eyepoints, otherwise just the active ones\n"
		print "\n"
		return

   # initialize the OpenFlight API
   # always call mgInit BEFORE any other OpenFlight API calls 
   #
	mgInit (None, None)

   # open the database 
	print "\nOpening database: ", sys.argv[1], " \n"
	db = mgOpenDb (sys.argv[1])
	if db == None:
		msgbuf = mgGetLastError()
		print msgbuf, "\n"
		mgExit()
		return

   # loop through the 9 eyepoints stored in database 
	for i in range (1, 10):
	
		eyePoint = mgGetEyePoint (db, i)
		if eyePoint != None:
			active = 0
			numAttr = mgGetAttList (eyePoint, fltEyeActive)
			if numAttr[0] != 1:
				print "Error getting eyepoint attribute\n"
			else:
				active = numAttr[2]
				if writeNonActives == MG_TRUE or active:
					PrintEyePoint (i, eyePoint)
			
 		
	

   # close the database 
	ok = mgCloseDb (db)
	if ok == MG_FALSE:
		print "Error closing database\n"

	# always call mgExit() AFTER all OpenFlight API calls 
	mgExit()

main()

File: workUndone/Suite15/OpenFlight_API/samples/apps/src/egeyepoint1.c
/*******************************************************************************

  Sample file: EGEYEPOINT1.C

  Program functions:	Opens a database from the command line.
     Reads in eyepoint records from the database
	  Prints out all eyepoints, or only active eyepoints in the database
	  Closes the database.

  API functions used:
     mgInit(), mgExit(), mgGetLastError(),
	  mgGetAttList(), mgGetEyePoint(), mgSetEyePoint(),
	  mgOpenDb(), mgCloseDb(), 

*******************************************************************************/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/* include all OpenFlight API headers */
#include <mgapiall.h>
#include "printeye.h"

int main (int argc, char* argv[])
{
	int i;
	mgrec* eyePoint;
	mgrec* db;
	/* whether to write out eyepoints that haven't been saved */
	mgbool writeNonActives = MG_FALSE; 
	mgbool ok;

	/* check for proper arguments */
	if ((argc == 3) && (strcmp(argv[2], "-a") == 0))
		writeNonActives = MG_TRUE;
	else if (argc < 2) {
		printf ("\nUsage: %s <input_db_filename> [-a]\n", argv[0]);
		printf ("   Reads database: <input_db_filename>\n");
		printf ("   Prints eyepoints from database\n");
		printf ("   -a : Print all eyepoints, otherwise just the active ones\n");
		printf ("\n");
		exit (EXIT_FAILURE);
	}

	/* initialize the OpenFlight API
		always call mgInit BEFORE any other OpenFlight API calls 
		*/
	mgInit (&argc, argv);

	/* open database */
   printf ("\nOpening database: %s\n", argv[1]);
	db = mgOpenDb (argv[1]);
	if (db == MG_NULL) {
		char msgbuf [1024];
		mgGetLastError (msgbuf, 1024);
		printf ("Unable to open database %s: %s", argv[1], msgbuf);
		mgExit ();
		exit (EXIT_FAILURE);
	}

	/* loop through the 9 eyepoints stored in database */
	for (i = 1; i <= 9; i++)
	{
		eyePoint = mgGetEyePoint (db, i);
		if (eyePoint != MG_NULL) {
			int active = 0;
			int numAttr = mgGetAttList (eyePoint, fltEyeActive, &active, MG_NULL);
			if (numAttr != 1)
				printf ("Error getting eyepoint attribute\n");
			else {
				if ((writeNonActives == MG_TRUE) || active)
					PrintEyePoint (i, eyePoint);
			}
 		}
	}

	/* Close the database and exit */
	ok = mgCloseDb (db);
	if (ok == MG_FALSE)
		printf ("Error closing database\n");

	/* always call mgExit() AFTER all API function calls */
	mgExit ();

   exit (0);
}
